src/Apis.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from "react";
+
+const useFlightApi = (url, id = null) => {
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+        const jsonData = await response.json();
+        setData(jsonData);
+        setLoading(true);
```

Comment:
`setLoading(true) should be set before the api is called. why is it set after the api call ?